### DISCUSSION TITLE
how to switch Kamon on/off ?

### DISCUSSION THREAD

**Discussion by JR-Utily** [kamon-io/Kamon#877] [Q&A] [answered]
Hello !
I'm wondering is there is a way to "switch" the instrumentation of Kamon on/off
We are considering using it in my team to have better logging through Futures etc
But in our mind, it would be like a DebugMode, that would do nothing if not activated.

So is there a live parameter somewhere to switch off the monitoring ?
Or is the fact not to add the agent a safe way to switch ?
Or should it be always on as soon as we include a Kamon.init() in our code ?

I've done a POC where it *seems* that not including Kanela agent is ok, but I'm not sure of the implication of that (as the note on traceBody() Future instrumentation says it is not safe).

And I've found this issue kamon-io/kanela#105 so I fear it is not the case.

**Comment by SimunKaracic** [comment, marked as answer]
Unfortunately, there's no setting right now to support something like that. 
We'd like to add it, but even when we add it, it's not going to be a "live" thing. 
The app will either start with Kamon turned on, or turned off.

For now, to truly turn off Kamon, you need to not call `Kamon.init()`, and if you were adding the kanela agent through the command line, remove it.
Anything else will leave Kamon in an unsafe, half loaded state, so you should not do that. 

Thanks for opening up the first discussion, please don't hesitate to ask more question :D :tada:
